eZ/Bundle/EzPublishIOBundle/Command/MigrateFilesCommand.php

```diff
@@ -0,0 +1,276 @@
+            if (count($fromHandlers) !== 2) {
+                $output->writeln('Enter two comma separated values for the --from option: <from_metadata_handler>,<from_binarydata_handler>');
+
+                return false;
+            }
+
+            if (!in_array($fromHandlers[0], array_keys($this->configuredMetadataHandlers))) {
+                $output->writeln("From meta data handler '$fromHandlers[0]' is not configured.");
+                $this->outputConfiguredHandlers($output);
+
+                return false;
+            }
+
+            if (!in_array($fromHandlers[1], array_keys($this->configuredBinarydataHandlers))) {
+                $output->writeln("From binary data handler '$fromHandlers[1]' is not configured.");
+                $this->outputConfiguredHandlers($output);
+
+                return false;
+            }
+        }
+
+        if ($toHandlers) {
+            if (count($toHandlers) !== 2) {
+                $output->writeln('Enter two comma separated values for the --to option: <to_metadata_handler>,<to_binarydata_handler>');
+
+                return false;
+            }
+
+            if (!in_array($toHandlers[0], array_keys($this->configuredMetadataHandlers))) {
+                $output->writeln("To meta data handler '$toHandlers[0]' is not configured.");
+                $this->outputConfiguredHandlers($output);
+
+                return false;
+            }
+
+            if (!in_array($toHandlers[1], array_keys($this->configuredBinarydataHandlers))) {
+                $output->writeln("To binary data handler '$toHandlers[1]' is not configured.");
+                $this->outputConfiguredHandlers($output);
+
+                return false;
+            }
+        }
+
+        if ($fromHandlers && $toHandlers && $fromHandlers === $toHandlers) {
+            $output->writeln('From and to handlers are the same. Nothing to do.');
+
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Migrate files.
+     *
+     * @param int $fileCount
+     * @param int $bulkCount
+     * @param bool $dryRun
+     * @param OutputInterface $output
+     */
+    protected function migrateFiles(
+        $fileCount,
+        $bulkCount,
+        $dryRun,
+        OutputInterface $output
+    ) {
+        $passCount = ceil($fileCount / $bulkCount);
+
+        $progress = new ProgressBar($output, $fileCount);
+        $progress->setFormat(' %current%/%max% [%bar%] %percent:3s%% %elapsed:6s%/%estimated:-6s% %memory:6s%');
+
+        $output->writeln('');
+        $progress->start();
+
+        for ($pass = 0; $pass <= $passCount; ++$pass) {
+            $metadataList = $this->migrationHandler->loadMetadataList($bulkCount, $pass * $bulkCount);
+
+            foreach ($metadataList as $metadata) {
+                if (!$dryRun) {
+                    $this->migrationHandler->migrateFile($metadata);
+                }
+
+                $progress->clear();
+
+                $output->write("\r");
```

Comment:
This is unfinished stuff. It will become obvious with larger datasets that this won't do, update coming soon. (It turns out that the ProgressBar supports all we need, including running without knowing the total count.)